Documentation/planning/arcade-powered-source-build/README.md

```diff
@@ -0,0 +1,160 @@
+     source-build tarball availability, as long as we treat a source-build
+     failure as seriously as a Microsoft build failure.
+
+2. PR validation. Every repo involved in source-build validates
+   source-buildability in its PR validation build.
+
+   * This allows developers and release owners to understand the source-build
+     impact of changes, reducing the frequency the source-build servicing team
+     has to root-cause and patch over problems.
+
+This doc is about where we can start, what incremental progress would look like,
+and the vision.
+
+## Starting point: official build
+
+The output of source-build is a set of tarballs that can be used to build the
+.NET Core SDK from source. We can move the current behavior of source-build to
+the Core-SDK official build. That is, Core-SDK clones all constituent repos,
+applies patches, builds each repo using customized build commands, and produces
+the source-build tarballs as artifacts.
+
+This immediately makes the dotnet/source-build repo unnecessary: it only held
+the source-build orchestration behavior.
+
+This needs more work to meet our goals due to build performance. It is simply
+too slow (> 2hrs) to build all constituent repos within one official build. It
+needs to be fast enough that it is reasonable for the entire official build to
+be rejected when the source-build fails.
+
+> Note: practically, the source-build official build should run in an
+> independent build pipeline at first: the long build time would interfere with
+> other .NET 5 work if integrated directly.
+
+For more info, see [source-build-in-pipeline.md].
+
+## Starting point: PR validation
+
+We can start here by adding extra jobs that run the standard source-build
+command and arguments. This is a simple step to confirm the build isn't
+fundamentally broken.
+
+This needs more work to meet our goals for many reasons:
+
+* Prebuilt dependency usage isn't tracked, because all dependencies are
+  downloaded as non-source-built prebuilts.
+* Source-build behavior may not work with source-built upstreams.
+* The artifacts built by the repo may not work downstream.
+* Advanced build flows aren't checked, such as source-build bootstrap or using
+  an N-1 SDK.
+
+For more info, see [source-build-in-pipeline.md].
+
+## Incremental progress
+
+### The performance gap
+We need to avoid building all constituent repos in the Core-SDK build. To do
+this, each repo needs to produce intermediate source-built artifacts during its
+official build, to be consumed by downstream repos. On the other end,
+source-build needs to support restoring from an intermediate artifact.
+
+To make incremental progress, we should pick one of Core-SDK's upstreams, and
+add source-build functionality that produces source-built intermediates.
+Core-SDK should consume them. We should choose a leaf in the source-build
+dependency graph, say, SourceLink. When Core-SDK is looking at the build graph
```

Comment:
Well, `aspnet-xdt` isn't a leaf here, since it uses Arcade. 🙂 But yeah, maybe `linker` would be more interesting since it has multiple direct downstreams.

Reply:
Hmm, I guess SourceLink is really double-weird since it uses a bootstrap version of Arcade instead of our version.  But yeah, we can just use whatever works and is interesting.